### PR TITLE
Add HTML lang attribute to application layout

### PR DESCRIPTION
**Pull request by MatheusRich**
### Motivation / Background

It is [recommended] to always have this attribute set on the html tag.

[recommended]: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/lang

### Detail

It adds the lang attribute based on the `I18n` locale.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: railties/lib/rails/generators/rails/app/templates/app/views/layouts/application.html.erb.tt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <!DOCTYPE html>
-<html>
+<html lang="<%= I18n.locale %>">
```
Comment (MatheusRich) [inline railties/lib/rails/generators/rails/app/templates/app/views/layouts/application.html.erb.tt:2]: Does this change need a CHANGELOG update/test?

File: railties/test/generators/app_generator_test.rb [hunk 2]
```diff
@@ -590,6 +590,14 @@ def test_does_not_generate_system_test_files_if_skip_system_test_is_given
     end
   end
 
+  def test_html_lang_attribute_is_present
+    run_generator [destination_root]
+
+    assert_file "app/views/layouts/application.html.erb" do |contents|
+      assert_match(/<html lang="en"/, contents)
```
Comment (skipkayhil) [inline railties/test/generators/app_generator_test.rb:597]: Based on this test, it looks like the intention is for the app to be generated with a static locale? If that's the case, how would a user configure which locale to generate statically?

Comment (MatheusRich) [reply railties/test/generators/app_generator_test.rb:597]: Whoops. Sorry, I got myself confused! I forgot to escape the ERB tag!

Comment (MatheusRich) [reply railties/test/generators/app_generator_test.rb]: Fixed!

**Comment by zzak** [comment]
Closing for reasons explained in #42404 and #43840

**Pull request closed without merging**